tutorials/dataframe/df106_HiggsToFourLeptons.py

```diff
@@ -18,32 +18,27 @@
 ##
 ## \date March 2020
 ## \author Stefan Wunsch (KIT, CERN)
+## \date May 2023
+## \author Marta Czurylo (CERN)
 
 import ROOT
-import json
 import os
 
-# Create a ROOT dataframe for each dataset
-# Note that we load the filenames from the external json file placed in the same folder than this script.
-path = "root://eospublic.cern.ch//eos/opendata/atlas/OutreachDatasets/2020-01-22"
-files = json.load(open(os.path.join(ROOT.gROOT.GetTutorialsDir(), "dataframe/df106_HiggsToFourLeptons.json")))
-
-processes = files.keys()
-df = {}
-xsecs = {}
-sumws = {}
-samples = []
-for p in processes:
-    for d in files[p]:
-        # Construct the dataframes
-        folder = d[0] # Folder name
-        sample = d[1] # Sample name
-        xsecs[sample] = d[2] # Cross-section
-        sumws[sample] = d[3] # Sum of weights
-        samples.append(sample)
-        df[sample] = ROOT.RDataFrame("mini", "{}/4lep/{}/{}.4lep.root".format(path, folder, sample))
+# Create the RDataFrame from the spec json file. The spec4lep.json is provided in the same folder as this tutorial.
+
+my_spec = os.path.join(ROOT.gROOT.GetTutorialsDir(), "dataframe", "df106_HiggsToFourLeptons_spec.json")
+
+df = ROOT.RDF.Experimental.FromSpec(my_spec) #creates a single dataframe for all the samples
+
+# Access metadata information that is stored in the json config file of the RDataFrame
+
+df = df.DefinePerSample("xsecs", 'rdfsampleinfo_.GetD("xsecs")') #GetD for "double"
```

Comment:
I don't think we should place comments like `GetD for double` and similar here. For the documentation users should refer to `RDFSampleInfo` or `RMetaData`. What maybe we can do is writing above the first call to `DefinePerSample` a comment similar to

```
# The metadata contained in the JSON file is accessible within a `DefinePerSample` call, through the `RDFSampleInfo` class
```